### ISSUE TITLE
Internal LoadBalancers do not release allocated IP addresses when service is deleted

### ISSUE THREAD

**Issue by fr6nco** [kubernetes-sigs/cloud-provider-azure#3226]
#### What happened:

Several services were exposed as an internal load balancer (annotated with service.beta.kubernetes.io/azure-load-balancer-internal: "true"). When these services were updated to type ClusterIP, private IP addresses were not released but present in the fontend IP configuration.

#### What you expected to happen:

Private IPs to be released and attachment removed from the Frontent IP configuration.

#### How to reproduce it (as minimally and precisely as possible):

Create a LoadBalancer Service with annotation service.beta.kubernetes.io/azure-load-balancer-internal: "true". Wait until fully provisioned (private IP assigned in azure console). Patch this Service, set type to ClusterIP and remove annotation. IP address is not released from the loadbalancer service. Perhaps try multiple loadbalancers with this annotation and change only one of them to ClusterIP and see if the updated resource deleted the IP address allocation.

#### Environment:

- Server Version: version.Info{Major:"1", Minor:"24", GitVersion:"v1.24.2", GitCommit:"f66044f4361b9f1f96f0053dd46cb7dce5e990a8", GitTreeState:"clean", BuildDate:"2022-06-15T14:15:38Z", GoVersion:"go1.18.3", Compiler:"gc", Platform:"linux/amd64"}
- AlmaLinux release 8.5 (Arctic Sphynx)
- Linux azure-madme-prod-k8s-master-0 4.18.0-348.2.1.el8_5.x86_64 #1 SMP Mon Nov 15 09:17:08 EST 2021 x86_64 x86_64 x86_64 GNU/Linux
- Installed via kubespray.
- Cloud provider installed via helm.

**Comment by feiskyer** [comment]
Do you share more details on "IP address is not released"? Do you mean the IP address is still on service status, or the LoadBalancer rule/frontend configurations are not deleted?

**Comment by fr6nco** [comment]
The frontend configuration is not deleted. The IP address was still attached to the Load Balancer in the azure console.

**Comment by fr6nco** [comment]
Issue still present on 1.25. 

Attaching some screenshots.


This was the old service definition
```
engine-ws-api-prod01-engine-ws-api-server            LoadBalancer   10.235.26.112   172.32.0.15                              80:30678/TCP,18291:31834/TCP                                                                                                 137d
```
which got changed to
```
[root@azure-ooredoo-prod-k8s-master-0 ~]# kubectl get svc -n prod engine-ws-api-prod01-engine-ws-api-server
NAME                                        TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)            AGE
engine-ws-api-prod01-engine-ws-api-server   ClusterIP   10.235.26.112   <none>        80/TCP,18291/TCP   137d
```

<img width="1386" alt="image" src="https://github.com/kubernetes-sigs/cloud-provider-azure/assets/1494663/5735c292-0189-426f-b9e3-5d94007108f6">

Associated IP address was not removed from the azure console tho.

![image](https://github.com/kubernetes-sigs/cloud-provider-azure/assets/1494663/3820e401-93f7-4df4-95d5-3a2b092e3e9b)

**Comment by nilo19** [comment]
Can you provide the logs of cloud-controller-manager during the service creation and deletion?

**Comment by fr6nco** [comment]
Seems to be fixed on 1.27+
